### PR TITLE
Remove widgets individually in the protocol

### PR DESCRIPTION
**Pull request by JakeWharton**
This is required to efficiently implement `movableContentOf` in a future change.

See #1902 

---

- [ ] `CHANGELOG.md`'s "Unreleased" section has been updated, if applicable.

### CODE REVIEW

File: redwood-protocol-guest/src/commonTest/kotlin/app/cash/redwood/protocol/guest/ProtocolTest.kt [hunk 1]
```diff
@@ -288,6 +291,46 @@ class ProtocolTest {
...
+      .containsExactly(
+        ChildrenChange.Remove(Id.Root, ChildrenTag.Root, 2, false),
+        ChildrenChange.Remove(Id.Root, ChildrenTag.Root, 1, false),
+        ChildrenChange.Remove(Id.Root, ChildrenTag.Root, 0, false),
+      )
+  }
+
+  @Suppress("DEPRECATION") // Testing for old behavior.
+  @Test
+  fun removeRangeForOldVersions() = runTest {
```
Comment (squarejesse) [inline redwood-protocol-guest/src/commonTest/kotlin/app/cash/redwood/protocol/guest/ProtocolTest.kt:305]: nice

File: redwood-treehouse-guest/src/jsMain/kotlin/app/cash/redwood/treehouse/ProtocolBridgeJs.kt [hunk 2]
```diff
@@ -208,7 +214,13 @@ internal class FastGuestProtocolAdapter(
     val tag = tag
     val index = index
     val count = count
-    changes.push(js("""["remove",{"id":id,"tag":tag,"index":index,"count":count}]"""))
+    if (hostSupportsRemoveDetachAndNoCount) {
```
Comment (squarejesse) [inline redwood-treehouse-guest/src/jsMain/kotlin/app/cash/redwood/treehouse/ProtocolBridgeJs.kt:217]: I assume in follow-up we'll change the caller to use a different function?

**Review by squarejesse — APPROVED** [review]

File: redwood-protocol-guest/src/commonTest/kotlin/app/cash/redwood/protocol/guest/ProtocolTest.kt (hunk 1 above)
Comment (JakeWharton) [reply redwood-protocol-guest/src/commonTest/kotlin/app/cash/redwood/protocol/guest/ProtocolTest.kt:305]: I'm actually thinking this behavior switching is pointless now. I broke this change out of the large one, but it's always valid to send individual removes to any host. So we always can do it and only worry about whether or not it's valid to send `detach=true` or not.

Comment (JakeWharton) [reply redwood-protocol-guest/src/commonTest/kotlin/app/cash/redwood/protocol/guest/ProtocolTest.kt:305]: Just going to leave it for now, land the rest which are kinda stacked on it, and then maybe delete.

Comment (JakeWharton) [reply redwood-protocol-guest/src/commonTest/kotlin/app/cash/redwood/protocol/guest/ProtocolTest.kt:305]: Yeah I got rid of this in #2509.

**Pull request merged**